### ISSUE TITLE
Feature request: add `ignore` option to `forbid-prop-types`

### ISSUE THREAD

**Issue by huchenme** [yannickcr/eslint-plugin-react#2025]
What do you think if we add `ignore` option to `forbid-prop-types` so that we could ignore certain props?

**Comment by ljharb** [comment]
Can you elaborate on the use case?

**Comment by huchenme** [comment]
For some props like `theme` which is related to styles, we don't really want to add all the possibilities of it. We could add `ignore` option to `prop-types` but we want to make sure `theme` prop is present, just through warning if I use

```
theme: Proptypes.object
```

Right not I need to hack ESlint using

```
theme: Proptypes.shape()
```

but I don't prefer to do it.

**Comment by ljharb** [comment]
I’d expect that you’d want to make a single shared `themePropType`, and that one would use PropTypes.object, or PropTypes.objectOf, etc, and use an eslint override if needed.

**Comment by huchenme** [comment]
@ljharb sorry not get you.

I want to forbid using `PropTypes.object` for all props except `theme`

**Comment by ljharb** [comment]
Right - what i mean is, you shouldn’t have a bunch of `PropTypes.object`s littered around your codebase - instead, make one file, `themePropType.js`:
```js
// eslint-disable-next-line react/forbid-prop-types
export default PropTypes.object; // or better, PropTypes.objectOf(something)
```

And then anywhere there’s a theme prop, `import themePropType from ‘./themePropType’`, and then `theme: themePropType`.

That way, if you ever want to change what a “theme” should be, you only have to do it in one place - and then you can use a single eslint-disable comment instead of needing to make your overarching rule config more complex.

**Comment by huchenme** [comment]
Thanks, this could work as well. But it will be nice if there is an `ignore` prop, since `react/prop-types` got `ignore` prop as well

**Comment by ljharb** [comment]
The problem here is that forbid-prop-types might have `a, b, c` - but you might only want to ignore `a` on prop `foo`. Configuring that is much more complex than a standalone "ignore" list.

Additionally, since your use case is to support an antipattern (repeating a prop type in multiple places, instead of putting it in one shared single source of truth), i'm not sure the complexity is warranted.

**Comment by ljharb** [comment]
I don't think allowing too-broad ignoring of forbidden propTypes is a good start - these things are forbidden because they're always terrible, and it's better to make ignoring them hard, then to make using them easier.
